bundles/core/src/test/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClientTest.java

```diff
@@ -14,106 +14,162 @@
 
 package com.adobe.cq.commerce.core.components.client;
 
+import java.io.IOException;
 import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Hashtable;
 import java.util.List;
 
 import org.apache.http.Header;
 import org.apache.http.message.BasicHeader;
 import org.apache.sling.api.resource.Resource;
+import org.apache.sling.caconfig.ConfigurationBuilder;
+import org.apache.sling.testing.mock.caconfig.ContextPlugins;
 import org.apache.sling.testing.mock.sling.ResourceResolverType;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.mockito.ArgumentMatcher;
 import org.mockito.Mockito;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.adobe.cq.commerce.graphql.client.GraphqlClient;
 import com.adobe.cq.commerce.graphql.client.HttpMethod;
 import com.adobe.cq.commerce.graphql.client.RequestOptions;
 import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
 import io.wcm.testing.mock.aem.junit.AemContext;
-import io.wcm.testing.mock.aem.junit.AemContextCallback;
+import io.wcm.testing.mock.aem.junit.AemContextBuilder;
 
 public class MagentoGraphqlClientTest {
-
+    private static final Logger LOG = LoggerFactory.getLogger(MagentoGraphqlClientTest.class);
     private GraphqlClient graphqlClient;
 
+    private ConfigurationBuilder mockConfigurationBuilder;
+
     @Rule
-    public final AemContext context = createContext("/context/jcr-content.json");
-
-    private static AemContext createContext(String contentPath) {
-        return new AemContext(
-            (AemContextCallback) context -> {
-                // Load page structure
-                context.load().json(contentPath, "/content");
-            },
-            ResourceResolverType.JCR_MOCK);
-    }
+    public final AemContext context = new AemContextBuilder(ResourceResolverType.JCR_MOCK).plugin(ContextPlugins.CACONFIG)
+        .beforeSetUp(context -> {
+            ConfigurationAdmin configurationAdmin = context.getService(ConfigurationAdmin.class);
+            Configuration serviceConfiguration = configurationAdmin.getConfiguration(
+                "org.apache.sling.caconfig.resource.impl.def.DefaultContextPathStrategy");
+
+            Dictionary<String, Object> props = new Hashtable<>();
```

Comment:
Can you document what this does?... and isn't there anything better to setup this test from `org.apache.sling.testing.caconfig-mock-plugin`? I expected that the content you added to `jcr-conf.json` would be enough ...

Reply:
The Sling CAConfig API uses different property names (`sling:configs` instead of `settings`,`sling:configRef` instead of `cq:conf` etc) so I configured the mock components as they are in AEM. Of course, I could've just used those Sling properties in the test JSON but I wanted the test content to be as close as possible to our AEM use case.

Reply:
Thanks for the info ... just out of curiosity, how is this configured then in AEM? I tried to find a config for `cq:conf` and that kind of things but couldn't find anything.

Reply:
(one eternity later...)

There is a bunch of `Apache Sling Context-Aware Configuration ...` components at http://localhost:4502/system/console/configMgr